### PR TITLE
feat(Telemetry): Add `commandOptionNames` to payload

### PR DESCRIPTION
**Pull request by pgrzesik**
Closes: #9385

### CODE REVIEW

File: lib/utils/telemetry/generatePayload.js [hunk 1]
```diff
@@ -119,12 +119,21 @@ module.exports = async (serverless = null) => {
     return resolveIsLocallyInstalled();
   })();
 
-  const { command } = resolveInput();
+  const { command, options, commandSchema } = resolveInput();
+
+  // We only consider options that are present in schema, with the exclusion of
+  // `region` and `stage`, because they are always present due to mutation of `options` object
```
Comment (medikoo) [inline lib/utils/telemetry/generatePayload.js:125]: I there's a mutation, let's fix it. I think we can easily fix it by ensure that we store copy of `options` in `Serverless` constructor.

**Review by medikoo — CHANGES_REQUESTED** [review]
Thanks @pgrzesik, Still I would definitely fix this not intended mutation so it doesn't leak to telemetry, and at telemetry ensure to report `region` and `stage` if user literally used those options in a CLI

File: lib/utils/telemetry/generatePayload.js (hunk 1 above)
Comment (pgrzesik) [reply lib/utils/telemetry/generatePayload.js]: That's a good call, fixed

File: lib/Serverless.js [hunk 2]
```diff
@@ -87,7 +87,8 @@ class Serverless {
       this._shouldReportMissingServiceDeprecation = true;
     }
     const commands = ensureArray(configObject.commands, { isOptional: true });
-    const options = ensurePlainObject(configObject.options, { isOptional: true });
+    let options = ensurePlainObject(configObject.options, { isOptional: true });
+    if (options) options = { ...options };
```
Comment (medikoo) [inline lib/Serverless.js:91]: Let's maybe add a comment on why we're doing that (technically it's a temporary hack until we have https://github.com/serverless/serverless/issues/2582 addressed)

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**